recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/InteractionFailureStrategy.scala

```diff
@@ -1,46 +1,57 @@
+                                                        private var backoffFactor: Double = 2,
+                                                        private var deadline: Option[java.time.Duration] = None,
+                                                        private var maximumRetries: Option[Int] = None,
+                                                        private var maxTimeBetweenRetries: Option[java.time.Duration] = None,
+                                                        private var fireRetryExhaustedEvent: Option[String] = None) {
+    def this() = this(None, 2, None, None, None, None)
+
+    def withInitialDelay(initialDelay: java.time.Duration) = this.copy(initialDelay = Some(initialDelay))
+
+    def withBackoffFactor(backoffFactor: Double) = this.copy(backoffFactor = backoffFactor)
+
+    def withMaximumRetries(maximumRetries: Int) = this.copy(maximumRetries = Some(maximumRetries))
+
+    def withMaxTimeBetweenRetries(maxTimeBetweenRetries: java.time.Duration) = this.copy(maxTimeBetweenRetries = Some(maxTimeBetweenRetries))
+
+    def withFireRetryExhaustedEvent(fireRetryExhaustedEvent: String) = this.copy(fireRetryExhaustedEvent = Some(fireRetryExhaustedEvent))
+
+    def withFireRetryExhaustedEvent(fireRetryExhaustedEvent: Boolean) = this.copy(fireRetryExhaustedEvent = Some(common.defaultEventExhaustedName))
+
+    def withFireRetryExhaustedEvent(fireRetryExhaustedEvent: Class[_]) = this.copy(fireRetryExhaustedEvent = Some(fireRetryExhaustedEvent.getSimpleName))
+
+    def withDeadline(deadline: java.time.Duration) = this.copy(deadline = Some(deadline))
+
+    def build(): RetryWithIncrementalBackoff = {
+      require(initialDelay.isDefined, "InitialDelay must be defined")
 
-  def RetryWithIncrementalBackoff(initialDelay: java.time.Duration,
-                                  deadline: java.time.Duration,
-                                  maxTimeBetweenRetries: java.time.Duration,
-                                  fireRetryExhaustedEvent: Boolean): RetryWithIncrementalBackoff = {
-    val initialDelayDuration = Duration(initialDelay.toMillis, duration.MILLISECONDS)
-    val deadlineDuration = Duration(deadline.toMillis, duration.MILLISECONDS)
-    common.InteractionFailureStrategy.RetryWithIncrementalBackoff(
-      initialDelayDuration,
-      deadlineDuration,
-      Some(Duration(maxTimeBetweenRetries.toMillis, duration.MILLISECONDS)),
-      fireRetryExhaustedEvent)
+      var builder: common.InteractionFailureStrategy.RetryWithIncrementalBackoff.builder = common.InteractionFailureStrategy.RetryWithIncrementalBackoff.builder()
+      if (deadline.isDefined)
+        builder = builder.withDeadline(Duration(deadline.get.toMillis, duration.MILLISECONDS))
+      else if (maximumRetries.isDefined)
+        builder = builder.withMaximumRetries(maximumRetries.get)
+      else {
+        throw new IllegalArgumentException("Either deadline of maximum retries need to be set")
+      }
```

Comment:
Question: Is it possible to set a deadline and maximumRetries? Or is it always one of those options?

Reply:
To elaborate the question; Can the strategy handle it if both deadline and maximumRetries are set. And is there a case that it makes sense?

Reply:
Only one can be truly set. Now if both are set we pick deadline but it would be better if we picked last set on the builder.